tests/_test_msui/test_multiple_flightpath_dockwidget.py

```diff
@@ -25,33 +25,230 @@
-        self.waypoints_model = ft.WaypointsTableModel("myname")
-        self.waypoints_model.insertRows(
-            0, rows=len(initial_waypoints), waypoints=initial_waypoints)
-
-        self.widget = tv.MSUITopViewWindow(model=self.waypoints_model, mainwindow=self.window)
-        self.window.show()
-        QtTest.QTest.qWaitForWindowExposed(self.window)
-        yield
-        self.window.hide()
-
-    def test_initialization(self):
-        widget = MultipleFlightpathControlWidget(parent=self.widget,
-                                                 listFlightTracks=self.window.listFlightTracks)
-        assert widget.color == (0, 0, 1, 1)
+@pytest.fixture
+def main_window(qtbot):
+    """
+    Set-up for the docking widget
+    """
+    # Start a MSUI window
+    window = msui.MSUIMainWindow()
+    window.show()
+    qtbot.wait_exposed(window)
+
+    # Create two flight tracks
+    window.actionNewFlightTrack.trigger()
+    window.actionNewFlightTrack.trigger()
+
+    # Open a Top View window
+    window.actionTopView.trigger()
+    topview_window = window.listViews.currentItem().window
+
+    # Switch to the Multiple Flightpath Widget
+    topview_window.cbTools.setCurrentIndex(6)
+
+    # Get a reference to the created MultipleFlightpathControlWidget
+    multiple_flightpath_widget = topview_window.docks[tv.MULTIPLEFLIGHTPATH].widget()
+
+    yield window, multiple_flightpath_widget
+
+    window.hide()
+
+
+def test_initialization(main_window):
+    """
+    test for conforming docking widget has initialized
+    """
+    _, multiple_flightpath_widget = main_window
+
+    # Ensure the MultipleFlightpathControlWidget is correctly initialized
+    assert multiple_flightpath_widget is not None
+    assert multiple_flightpath_widget.color == (0, 0, 1, 1)
+
+
+def test_setColour(main_window):
+    """
+    test for the filghttrack colour
+    """
+    main_window, multiple_flightpath_widget = main_window
+    color_button = multiple_flightpath_widget.pushButton_color
+
+    # Mock the exec_ method to accept the dialog otherwise test get stuck
+    with mock.patch("mslib.utils.colordialog.CustomColorDialog.exec_",
+                    return_value=QtWidgets.QDialog.Accepted) as mock_exec:
+        # Activate the first flight track
+        activate_flight_track_at_index(main_window, 0)
+
+        # Click on the second flight track in the docking widget
+        click_on_flight_track_in_docking_widget_at_index(multiple_flightpath_widget, 1)
+
+        # Simulate clicking the button to open the color dialog
+        color_button.click()
+        assert mock_exec.call_count == 1
+
+        color_dialog = multiple_flightpath_widget.findChild(QtWidgets.QDialog)
+        assert color_dialog is not None
+
+        # Simulate a color being selected
+        color = QtGui.QColor("#0000ff")  # Example color
+        color_dialog.color_selected.emit(color)
```

Comment:
My above suggestion would eliminate this signal emission, which I think doesn't belong into a test anyway. Completely mocking the exec_ method and directly emitting this signal breaks encapsulation and entirely skips a significant path through the color dialog and how it is hooked up with the MFDW, which I think is undesirable in a test.